### PR TITLE
Updates for 0.7

### LINKED ISSUE #11: update to 0.7

### PR DESCRIPTION
**Pull request by cryogenian**
fixes #11
- extracted ffi
- switched from `Number` to `Int` in most functions 
- replaced `[a]` -> `Array a`. But not switched from `Array a` to `List a` in public functions: is it necessary?

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -17,11 +17,12 @@
...
-    "purescript-arrays": "~0.3.0",
-    "purescript-foreign": "~0.4.0",
-    "purescript-simple-dom": "~0.0.2",
-    "purescript-nullable": "~0.1.0"
+    "purescript-arrays": "~0.4.0",
+    "purescript-foreign": "~0.5.0",
+    "purescript-nullable": "~0.2.0",
+    "purescript-simple-dom": "6358414013eb3106d695e3e19488609f82d903e9",
+    "purescript-prelude": "~0.1.0",
+    "purescript-console": "~0.1.0"
```
Comment (garyb) [inline bower.json]: I'd recommend using `^` ranges rather than `~` ranges for all versions now.

**Comment by garyb** [comment]
Switching from `Array` to `List` is definitely not a requirement - it depends on the use case. If there's a lot of `cons` and `uncons` going on then we probably want a `List`, if usage is much more convenient with `[]` literals than we probably want an `Array`, etc.

**Comment by cryogenian** [comment]
Switched from `~` to `^`

**Comment by jdegoes** [comment]
Looks good, can you squash commits?

**Comment by cryogenian** [comment]
squashed

**Pull request merged**